pipelines/pr.yml

```diff
@@ -86,39 +126,62 @@ resources:
   source: {repository: vito/oci-build-task}
 
 jobs:
-- name: unit
+- name: pr-image
   public: true
-  max_in_flight: 2
-  on_failure:
-    put: concourse-pr
-    inputs: [concourse-pr]
-    params: {path: concourse-pr, status: failure, context: unit}
+  serial: true
+  plan:
+  - in_parallel:
+    - get: concourse-pr
+      trigger: true
+      tags: [pr]
+    - get: oci-build-task
+      tags: [pr]
+    - get: dev-image
+      params: {format: oci}
+      tags: [pr]
+    - get: ci
+      tags: [pr]
+  - task: yarn-build
+    attempts: 3
+    file: ci/tasks/yarn-build.yml
+    input_mapping: {concourse: concourse-pr}
     tags: [pr]
-    get_params: {skip_download: true}
-  on_success:
-    put: concourse-pr
-    inputs: [concourse-pr]
-    params: {path: concourse-pr, status: success, context: unit}
+  - task: build
+    image: oci-build-task
+    privileged: true
+    config:
+      platform: linux
+      params:
+        CONTEXT: built-concourse
+        IMAGE_ARG_base_image: dev-image/image.tar
+        TARGET: with-fly
+      inputs: [{name: built-concourse}, {name: dev-image}]
+      outputs: [{name: image}]
+      # TODO: now that this is per PR pipeline, caches might end up taking a
+      # ton of disk...figure out if this is a problem.
+      caches: [{path: cache}]
```

Comment:
This is what I was talking about in retro. I think without some sort of GC mechanism for task caches we'll run into issues, since the size of the `pr` instance group never actually shrinks (old PR pipelines will get archived, but it seems as though task caches remain for archived pipelines)

Reply:
No longer blocked (see https://github.com/concourse/concourse/pull/7272)